frontend/src/components/header/__tests__/Header.test.js

```diff
@@ -1,59 +1,115 @@
-    const div = document.createElement('div')
-    ReactDOM.render(wrapper, div)
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H1>{example}</H1>
-      </ThemeProvider>,
-      div,
-    )
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H2>{example}</H2>
+import { render, cleanup } from 'react-testing-library'
+import { mount } from 'enzyme'
+import { Header, H1, H2, H3, H4, H5, H6 } from '../'
+
+describe('<Header />', () => {
+  afterEach(cleanup)
+
+  it('Properly render child components', () => {
+    const { getAllByText } = render(<Header is="h1">foo</Header>)
+
+    const test = getAllByText(/foo/)
+    expect(test).toHaveLength(1)
+  })
+
+  it('Properly handles font size props', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={{ fontSizes: ['14px'] }}>
+        <Header fontSize={0} is="h1">
+          foo
+        </Header>
       </ThemeProvider>,
-      div,
     )
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H3>{example}</H3>
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('font-size', '14px')
+  })
+
+  it('Properly handles line height props', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={{ lineHeights: ['1'] }}>
+        <Header lineHeight={0} is="h1">
+          foo
+        </Header>
       </ThemeProvider>,
-      div,
     )
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H4>{example}</H4>
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('line-height', '1')
+  })
+
+  it('Properly handles space props', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={{ space: ['5px', '10px'] }}>
+        <Header p={0} ml={1} is="h1">
+          foo
+        </Header>
       </ThemeProvider>,
-      div,
     )
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H5>{example}</H5>
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('padding', '5px')
+    expect(test).toHaveStyleRule('margin-left', '10px')
+  })
+
+  it('Properly handles color props', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={{ colors: { blue: '#005ea5' } }}>
+        <Header color="blue" is="h1">
+          foo
+        </Header>
       </ThemeProvider>,
-      div,
     )
-    ReactDOM.render(
-      <ThemeProvider theme={theme}>
-        <H6>{example}</H6>
-      </ThemeProvider>,
-      div,
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('color', '#005ea5')
+  })
+
+  it('Properly handles font weight', () => {
+    const { getByText } = render(
+      <Header fontWeight="bold" is="h1">
+        foo
+      </Header>,
     )
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('font-weight', 'bold')
+  })
+
+  it("The 'is' prop sets correct header tag", () => {
+    let wrapper = mount(<Header is="h1" />).find('h1')
```

Comment:
Do you need all these `find`s here and in subsequent tests? Isn't wrapper just going to be the entire thing anyway?

Reply:
If `shallow` rendering yes. Unfortunately I can't seem to use Shallow rendering for these? :( So I used `mount` which wraps it in a bunch of divs. So I used the finds to pick up the header

Reply:
wait a sec, why are you using `enzyme` here, can't something from `react-testing-library` do the job?

Reply:
I spent a while reading docs and looking things up, but was unable to find a matcher to check for HTML tag type. Enzyme had one so I decided to just use that